robot/rospackages/src/mcu_control/scripts/MockPdsNode.py

```diff
@@ -0,0 +1,199 @@
+        if (remainder > 0):
+            valueBeforeNoise = valueBeforeIncrement + increment #Increment the value presented on the gui
+            valueAfterNoise = round(valueBeforeNoise + (noise * (random.uniform(-1, 1))), 2) #Add noise to the incremented value
+        else:
+            valueBeforeNoise = valueBeforeIncrement - increment #Decrement the value presented on the gui
+            valueAfterNoise = round(valueBeforeNoise + (noise * (random.uniform(-1, 1))), 2) #Add noise to the incremented value
+    else:
+        valueAfterNoise = round(currentGetValue + (noise * (random.uniform(-1, 1))), 2) #Add noise to the value if it is not incrementing or decrementing
+
+    return valueAfterNoise
+
+def get_states_values(parameters, values):
+    """Initialize parameters with the values according to the states (rise, stable and fall)"""
+
+    #Check if parameters is a string (single parameter) or an array (multiple parameters)
+    if isinstance(parameters, str):
+        paramState = get_param_exist(parameters)
+        newValue = None
+        if (paramState == "rise"):
+            newValue = values.get("rise")
+        elif (paramState == "stable"):
+            newValue = values.get("stable")
+        elif (paramState == "fall"):
+            newValue = values.get("fall")
+        else:
+            rospy.logwarn("Failed: Invalid parameter entered")
+            #If noise or riseFallRate were never initialized in the launch file, set them to stable
+            if ((parameters == "PDS_mock_rise_fall_rate" or parameters == "PDS_mock_global_noise") and paramState == None):
+                    newValue = values.get("stable")
+                    rospy.logwarn("Noise or rate set to STABLE")
+        return newValue
+
+    #If parameters is an array, return the value associated for each parameter state
+    else:
+        outputArray = []
+        for x in range(len(parameters)):
+            paramState = get_param_exist(parameters[x])
+            if (paramState == "rise"):
+                outputArray.append(values.get("rise"))
+            elif (paramState == "stable"):
+                outputArray.append(values.get("stable"))
+            elif (paramState == "fall"):
+                outputArray.append(values.get("fall"))
+            else:
+                rospy.logwarn("Failed: Invalid parameter entered for {}".format(parameters[x]))
+        return outputArray
+
+def get_param_exist(parameter):
+    """Get parameter state if parameter exists and sets to None if it does not exist"""
+
+    try:
+        parameterState = rospy.get_param(parameter)
+    except:
+        rospy.logwarn("Parameter state None assigned to {}.".format(parameter))
+        parameterState = None
+    return parameterState
+
+def publish_mock_data(voltages, temps, currents, noiseValues, riseFallRates):
+    """Get and set parameters to be published"""
+
+    #Initialize node
+    node_name = "mock_pds_node"
+    rospy.init_node(node_name, anonymous=True)
```

Comment:
```suggestion
    rospy.init_node(node_name)
```

This should prevent people from running 2 instances of this node